x-pack/apm-server/sampling/eventstorage/storage_manager.go

```diff
@@ -164,15 +164,18 @@ func (s *StorageManager) runDropLoop(stopping <-chan struct{}, ttl time.Duration
 	var firstExceeded time.Time
 	checkAndFix := func() error {
 		lsm, vlog := s.Size()
-		// add buffer to avoid edge case storageLimitInBytes-lsm-vlog < buffer, when writes are still always rejected
-		buffer := int64(baseTransactionSize * len(s.rw.readWriters))
-		if uint64(lsm+vlog+buffer) >= storageLimitInBytes {
+		s.mu.RLock() // s.storage requires mutex RLock
+		pending := s.storage.pendingSize.Load()
```

Comment:
[to reviewer] This is just a nice to have to rule out any possibility that the condition is inconsistent with what's used in read writer.